ntoskrnl/include/internal/ps_i.h

```diff
@@ -27,7 +27,9 @@ static const INFORMATION_CLASS_INFO PsProcessInfoClass[] =
     (
         QUOTA_LIMITS,
         ULONG,
-        ICIF_QUERY | ICIF_SET | ICIF_SET_SIZE_VARIABLE
+
+        /* NOTE: ICIF_SIZE_VARIABLE is for QUOTA_LIMITS_EX support */
```

Comment:
So even for the query it needs the size-variable?

Reply:
Yes, otherwise `DefaultQueryInfoBufferCheck` will fail with `STATUS_INFO_LENGTH_MISMATCH` if `ProcessInformationLength != sizeof(QUOTA_LIMITS)`